grafonnet/bar_gauge_panel.libsonnet

```diff
@@ -0,0 +1,39 @@
+{
+  /**
+   * Create a [bar gauge panel](https://grafana.com/docs/grafana/latest/panels/visualizations/bar-gauge-panel/),
+   *
+   * @name barGaugePanel.new
+   *
+   * @param title Panel title.
+   * @param datasource Panel datasource.
+   * @param unit The unit of the data.
+   *
+   * @method addTarget(target) Adds a target object.
+   * @method addTargets(targets) Adds an array of targets.
+   */
+  new(
+    title,
+    datasource=null,
+    unit=null
+  ):: {
+    title: title,
+    datasource: datasource,
+    type: 'bargauge',
+    targets: [
+    ],
+    fieldConfig: {
+      defaults: {
+        unit: unit,
+      },
+    },
+    _nextTarget:: 0,
+    addTarget(target):: self {
+      // automatically ref id in added targets.
+      // https://github.com/kausalco/public/blob/master/klumps/grafana.libsonnet
```

Comment:
This link can go since it's not public or doesn't exist. It's in a few places though I know 😅